core/src/services/aliyun_drive/writer.rs

```diff
@@ -44,162 +43,89 @@ impl AliyunDriveWriter {
             _op: op,
             parent_file_id: parent_file_id.to_string(),
             name: name.to_string(),
-            file_id: Arc::new(RwLock::new(None)),
+            file_id: None,
+            upload_id: None,
+            part_number: 1, // must start from 1
         }
     }
 
-    async fn write_file_id(&self, id: String) {
-        let mut file_id = self.file_id.write().await;
-
-        *file_id = Some(id);
-    }
-
-    async fn read_file_id(&self) -> Result<String> {
-        let file_id = self.file_id.read().await;
-        let Some(ref file_id) = *file_id else {
-            return Err(Error::new(ErrorKind::Unexpected, "cannot find file_id"));
-        };
-
-        Ok(file_id.clone())
-    }
-
-    async fn write(
-        &self,
-        body: Option<Buffer>,
-        upload_url: Option<&str>,
-    ) -> Result<Option<String>> {
-        if let Some(upload_url) = upload_url {
-            let Some(body) = body else {
+    async fn get_upload_info(&mut self, do_not_create: bool) -> Result<(String, String)> {
```

Comment:
Hi, I think this function might be overdesigned. Could we simplify it by using `let Some(file_id) = &self.file_id {}`?

Reply:
make sense